src/test/regress/spec/isolation_citus_pause_node.spec

```
@@ -0,0 +1,219 @@
+	BEGIN
+		select nextval('pg_dist_node_nodeid_seq')::int into v_node_id;
+		select citus_pause_node_within_txn(v_node_id) ;
+	EXCEPTION
+		WHEN  SQLSTATE 'P0002' THEN
+			GET STACKED DIAGNOSTICS v_exception_message = MESSAGE_TEXT;
+			v_expected_exception_message := 'node ' || v_node_id || ' not found';
+			if v_exception_message = v_expected_exception_message then
+				RAISE NOTICE 'Node not found.';
+			end if;
+	END;
+	$$
+	LANGUAGE plpgsql;
+}
+
+step "s1-pause-node"
+{
+	SET client_min_messages = 'notice';
+	DO $$
+	DECLARE
+		v_shard_id int;
+		v_node_id int;
+		v_node_name text;
+		v_node_port int;
+	BEGIN
+		--The first message in the block is being printed on the top of the code block. So adding a dummy message
+		--to make sure that the first message is printed in correct place.
+		raise notice '';
+		-- Get the shard id for the distribution column
+		SELECT get_shard_id_for_distribution_column('employee', 3) into v_shard_id;
+
+		--Get the node id for the shard id
+		SELECT nodename,nodeport into v_node_name,v_node_port FROM citus_shards WHERE shardid = v_shard_id limit 1;
+		raise notice 'node name is %',v_node_name;
+		raise notice 'node port is %',v_node_port;
+
+		-- Get the node id for the shard id
+		SELECT nodeid into v_node_id FROM pg_dist_node WHERE nodename = v_node_name and nodeport = v_node_port limit 1;
+
+
+		-- Pause the node
+		perform pg_catalog.citus_pause_node_within_txn(v_node_id) ;
+	END;
+	$$
+	LANGUAGE plpgsql;
+}
+
+step "s1-pause-node-force"
+{
+	SET client_min_messages = 'notice';
+	DO $$
+	DECLARE
+		v_shard_id int;
+		v_node_id int;
+		v_node_name text;
+		v_node_port int;
+		v_force boolean := true;
+	BEGIN
+		--The first message in the block is being printed on the top of the code block. So adding a dummy message
+		--to make sure that the first message is printed in correct place.
+		raise notice '';
+		-- Get the shard id for the distribution column
+		SELECT get_shard_id_for_distribution_column('employee', 3) into v_shard_id;
+
+		--Get the node id for the shard id
+		SELECT nodename,nodeport into v_node_name,v_node_port FROM citus_shards WHERE shardid = v_shard_id limit 1;
+		raise notice 'node name is %',v_node_name;
+		raise notice 'node port is %',v_node_port;
+
+		-- Get the node id for the shard id
+		SELECT nodeid into v_node_id FROM pg_dist_node WHERE nodename = v_node_name and nodeport = v_node_port limit 1;
+
+
+		-- Pause the node with force true
+		perform pg_catalog.citus_pause_node_within_txn(v_node_id,v_force) ;
+	END;
+	$$
+	LANGUAGE plpgsql;
+}
+
+step "s1-end"
+{
+    COMMIT;
+}
+
+session "s2"
+
+
+step "s2-begin"
+{
+	BEGIN;
+}
+
+step "s2-insert-distributed"
+{
+	-- Set statement_timeout for the session (in milliseconds)
+	SET client_min_messages = 'notice';
+
+	-- Variable to track if the INSERT statement was successful
+	DO $$
+	BEGIN
+
+		-- Execute the INSERT statement
+		insert into employee values(11,'e11',3);
+
+	END;
+
+	$$
+	LANGUAGE plpgsql;
```

This insert step can now contain only a single insert statement afaict (same for the insert-reference and delete-distributed step below).

```suggestion
	insert into employee values(11,'e11',3);
```

---

Fixed